tokio/src/runtime/handle.rs

```diff
@@ -341,15 +341,37 @@ cfg_metrics! {
 cfg_taskdump! {
     impl Handle {
         /// Capture a snapshot of this runtime's state.
-        pub fn dump(&self) -> crate::runtime::Dump {
+        pub async fn dump(&self) -> crate::runtime::Dump {
             match &self.inner {
                 scheduler::Handle::CurrentThread(handle) => handle.dump(),
                 #[cfg(all(feature = "rt-multi-thread", not(tokio_wasi)))]
-                scheduler::Handle::MultiThread(_) =>
-                    unimplemented!("taskdumps are unsupported on the multi-thread runtime"),
+                scheduler::Handle::MultiThread(handle) => {
+                    // perform the trace in a separate thread so that the 
+                    // trace itself does not appear in the taskdump.
+                    let handle = handle.clone();
+                    spawn_thread(|| {
+                        let handle = handle;
+                        handle.dump()
+                    }).await
+                },
             }
         }
     }
+
+    /// Spawn a new thread and asynchronously await on its result.
+    async fn spawn_thread<F, T>(f: F) -> T
+    where
+        F: FnOnce() -> T + Send + 'static,
+        T: Send + 'static
+    {
+        let (tx, rx) = crate::sync::oneshot::channel();
+        crate::loom::thread::spawn(|| {
+            if tx.send(f()).is_err() {
+                unreachable!("send failed");
+            }
```

Comment:
You can trigger this by cancelling the operation. I would just ignore errors here.

Reply:
Resolved in https://github.com/tokio-rs/tokio/pull/5717/commits/4d97e1c1ce2d118e56cbb0c30ffbf5dd11846605#diff-f1d42d2af51248e5dacb1cc3dd6bba12a956abba0d25567ee67bfd68d27e27d5